dropshot/tests/test_streaming.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 Oxide Computer Company
+
+//! Test cases for streaming reqeusts.
+
+use dropshot::{endpoint, ApiDescription, HttpError, RequestContext};
+use http::{Method, Response, StatusCode};
+use hyper::{body::HttpBody, Body};
+use hyper_staticfile::FileBytesStream;
+use std::sync::Arc;
+use tokio::io::{AsyncSeekExt, AsyncWriteExt};
+
+#[macro_use]
+extern crate slog;
+
+mod common;
+
+fn api() -> ApiDescription<usize> {
+    let mut api = ApiDescription::new();
+    api.register(api_streaming).unwrap();
+    api
+}
+
+const BUF_SIZE: usize = 8192;
+const BUF_COUNT: usize = 128;
+
+#[endpoint {
+    method = GET,
+    path = "/streaming",
+}]
+async fn api_streaming(
+    _rqctx: Arc<RequestContext<usize>>,
+) -> Result<Response<Body>, HttpError> {
+    let mut file = tempfile::tempfile()
+        .map_err(|_| HttpError::for_bad_request(None, "EBADF".to_string()))
+        .map(|f| tokio::fs::File::from_std(f))?;
+
+    // Fill the file with some arbitrary contents.
+    let mut buf = [0; BUF_SIZE];
+    for i in 0..BUF_COUNT {
+        file.write_all(&buf).await.unwrap();
+        buf.fill((i & 255) as u8);
+    }
+    file.seek(std::io::SeekFrom::Start(0)).await.unwrap();
+
+    let file_stream = FileBytesStream::new(file);
+    Ok(Response::builder()
+        .status(StatusCode::OK)
+        .body(file_stream.into_body())?)
+}
+
+fn check_has_chunked_headers(response: &Response<Body>) {
+    let transfer_encoding_header =
+        response.headers().get("transfer-encoding").expect(
```

Comment:
We maybe want to check that this header is _not_ set when we're not streaming.  I guess that's implied if it's not one of the allowed headers already?

Reply:
Sure, I'll add another test with a non-streaming endpoint.